operations/admin.go

```diff
@@ -476,3 +478,100 @@ func updateRoleCmd() cli.Command {
 		},
 	}
 }
+
+func adminDistroExecute() cli.Command {
+	const (
+		distroFlagName            = "distro"
+		scriptPathFlagName        = "file"
+		scriptFlagName            = "script"
+		includeSpawnHostsFlagName = "include_spawn_hosts"
+		includeTaskHostsFlagName  = "include_task_hosts"
+		sudoFlagName              = "sudo"
+		sudoUserFlagName          = "sudo_user"
+	)
+	return cli.Command{
+		Name:  "distro-execute",
+		Usage: "run a shell script on selected hosts in a distro",
+		Flags: []cli.Flag{
+			cli.StringFlag{
+				Name:  distroFlagName,
+				Usage: "the distro to run the script on",
+			},
+			cli.StringFlag{
+				Name:  scriptFlagName,
+				Usage: "the script to run",
+			},
+			cli.StringFlag{
+				Name:  scriptPathFlagName,
+				Usage: "the file containing the script to run",
+			},
+			cli.BoolTFlag{
+				Name:  includeTaskHostsFlagName,
+				Usage: "run the script on hosts running tasks",
+			},
+			cli.BoolFlag{
+				Name:  includeSpawnHostsFlagName,
+				Usage: "run the script on spawn hosts",
+			},
+			cli.BoolFlag{
+				Name:  sudoFlagName,
+				Usage: "run the script with sudo",
+			},
+			cli.StringFlag{
+				Name:  sudoUserFlagName,
+				Usage: "run the script as a user",
+			},
+		},
+		Before: mergeBeforeFuncs(
+			requireStringFlag(distroFlagName),
+			mutuallyExclusiveArgs(true, scriptFlagName, scriptPathFlagName),
+		),
+		Action: func(c *cli.Context) error {
+			distro := c.String(distroFlagName)
+			includeTaskHosts := c.BoolT(includeTaskHostsFlagName)
+			includeSpawnHosts := c.Bool(includeSpawnHostsFlagName)
+			script := c.String(scriptFlagName)
+			sudo := c.Bool(sudoFlagName)
+			sudoUser := c.String(sudoUserFlagName)
+			if sudoUser != "" {
+				sudo = true
+			}
+			if script == "" {
+				scriptPath := c.String(scriptPathFlagName)
+				b, err := ioutil.ReadFile(scriptPath)
+				if err != nil {
+					return errors.Wrapf(err, "could not read script file '%s'", scriptPath)
+				}
+				script = string(b)
+			}
+
+			confPath := c.Parent().Parent().String(confFlagName)
+			conf, err := NewClientSettings(confPath)
+			if err != nil {
+				return errors.Wrap(err, "problem loading configuration")
+			}
+			client := conf.setupRestCommunicator(context.Background())
```

Comment:
why does this have a different context than 556?

Reply:
Fixed.